libp2p/RLPxHandshake.cpp

```diff
@@ -72,9 +99,19 @@ void RLPXHandshake::writeAck()
 	});
 }
 
+void RLPXHandshake::setAuthValues(Signature _sig, Public _remotePubk, h256 _remoteNonce, uint64_t _remoteVersion)
+{
+	_remotePubk.ref().copyTo(m_remote.ref());
+	_remoteNonce.ref().copyTo(m_remoteNonce.ref());
+	m_remoteVersion = _remoteVersion;
+	Secret sharedSecret;
+	crypto::ecdh::agree(m_host->m_alias.sec(), _remotePubk, sharedSecret);
+	m_remoteEphemeral = recover(_sig, sharedSecret.makeInsecure() ^ _remoteNonce);
```

Comment:
Perhaps being overly paranoid here: Is it possible to compute the `^` on `sharedSecret`'s data?